src/lib.rs

```diff
@@ -1021,6 +1046,338 @@ safety_comment! {
     unsafe_impl_known_layout!(T: ?Sized + KnownLayout => #[repr(T::MaybeUninit)] MaybeUninit<T>);
 }
 
+mod split_at {
+    use super::*;
+
+    /// Types that can be split in two.
+    ///
+    /// # Implementation
+    ///
+    /// **Do not implement this trait yourself!** Instead, use
+    /// [`#[derive(SplitAt)]`][derive]; e.g.:
+    ///
+    /// ```
+    /// # use zerocopy_derive::{SplitAt, KnownLayout};
+    /// #[derive(SplitAt, KnownLayout)]
+    /// #[repr(C)]
+    /// struct MyStruct<T: ?Sized> {
+    /// # /*
+    ///     ...,
+    /// # */
+    ///     field: T,
+    /// }
+    /// ```
+    ///
+    /// This derive performs a sophisticated, compile-time safety analysis to
+    /// determine whether a type is `SplitAt`.
+    ///
+    /// # Safety
+    ///
+    /// This trait does not convey any safety guarantees to code outside this crate.
+    ///
+    /// You must not rely on the `#[doc(hidden)]` internals of `SplitAt`. Future
+    /// releases of zerocopy may make backwards-breaking changes to these items,
+    /// including changes that only affect soundness, which may cause code which
+    /// uses those items to silently become unsound.
+    ///
+    #[cfg_attr(feature = "derive", doc = "[derive]: zerocopy_derive::SplitAt")]
+    #[cfg_attr(
+        not(feature = "derive"),
+        doc = concat!("[derive]: https://docs.rs/zerocopy/", env!("CARGO_PKG_VERSION"), "/zerocopy/derive.SplitAt.html"),
+    )]
+    #[cfg_attr(
+        zerocopy_diagnostic_on_unimplemented_1_78_0,
+        diagnostic::on_unimplemented(note = "Consider adding `#[derive(SplitAt)]` to `{Self}`")
+    )]
+    // SAFETY: The trailing slice is well-aligned for its element type.
+    pub unsafe trait SplitAt: KnownLayout<PointerMetadata = usize> {
+        /// The element type of the trailing slice.
+        type Elem;
+
+        // The `Self: Sized` bound makes it so that `SplitAt` is still object
+        // safe.
+        #[doc(hidden)]
+        fn only_derive_is_allowed_to_implement_this_trait()
+        where
+            Self: Sized;
```

Comment:
Add a test for object safety (where `Elem` is specified as a concrete type)?

Reply:
It's not object safe; comment removed.